pilot/pkg/serviceregistry/serviceentry/controller.go

```diff
@@ -515,6 +517,16 @@ func (s *Controller) WorkloadInstanceHandler(wi *model.WorkloadInstance, event m
 		} else {
 			s.serviceInstances.updateServiceEntryInstancesPerConfig(seNamespacedName, key, instance)
 		}
+		if se.Resolution == networking.ServiceEntry_DNS && se.WorkloadSelector != nil {
```

Comment:
Can you take a look at `workloadEntryHandler` ,  also consider networking.ServiceEntry_DNS_ROUND_ROBIN

Reply:
I mean `workloadEntryHandler ` has processed the issue here, you can refer the code there

Reply:
done